sim/rogue/item_sets_phase_8.go

```diff
@@ -1,6 +1,113 @@
+		return
+	}
+
+	label := "S03 - Item - Scarlet Enclave - Rogue - Damage 2P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	totalBleedsAndPoisons := rogue.PoisonsActive + rogue.BleedsActive
+
+	// TODO: Fix logic below here, checks above should be good. Using Feral Druid T2 6pc as reference here to start.
+	// Added bleed tracking variables much like Feral Druid, have updated Rupture, CT, Garrote, and SSL to add bleed trackers, need to figure out if Hemorrhage should count
+	// Testing done on 3/28/2025 in Classic Rogue Discord shows that using Luffa, Hemorrhage does not count as a bleed and should also not count towards this set bonus.
+	// Wowhead seems to confirm this as Hemorrhage does not have a Mechanic of Bleeding like Rupture, Garrote, etc
+	damageMod := rogue.AddDynamicMod(core.SpellModConfig{
+		Kind:       core.SpellMod_DamageDone_Pct,
+		ClassMask:  ClassSpellMask_RogueBackstab | ClassSpellMask_RogueSinisterStrike | ClassSpellMask_RogueSaberSlash | ClassSpellMask_RogueMutilate,
+		FloatValue: 1.0,
+	})
+
+	core.MakeProcTriggerAura(&rogue.Unit, core.ProcTrigger{
+		ActionID:       core.ActionID{SpellID: 1226843}, // Tracking in APL
+		Name:           label,
+		Callback:       core.CallbackOnApplyEffects,
+		ProcChance:     0.1,
+		ClassSpellMask: ClassSpellMask_RogueBackstab | ClassSpellMask_RogueSinisterStrike | ClassSpellMask_RogueSaberSlash | ClassSpellMask_RogueMutilate,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			fmt.Println("Bleeds: ", rogue.BleedsActive)
+			fmt.Println("Poisons: ", rogue.PoisonsActive)
+
+			// Only apply the damage mod up to 3 times for the 30% bonus maximum
+			fmt.Println("2P proc triggered")
+			if totalBleedsAndPoisons <= 3 {
+				fmt.Println("Applying 2P damage mod, value: ", totalBleedsAndPoisons)
+				damageMod.Activate()
+				damageMod.UpdateFloatValue(1 + 0.10*float64(totalBleedsAndPoisons))
+			}
+		},
+	})
+}
+
+// Your Poison and autoattack critical strikes have a 10% chance to grant you a combo point. (Proc chance: 10%)
+func (rogue *Rogue) applyScarletEnclaveDamage4PBonus() {
+	label := "S03 - Item - Scarlet Enclave - Rogue - Damage 4P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	comboPointMetrics := rogue.NewComboPointMetrics(core.ActionID{SpellID: 1226869})
+
+	// TODO: Figure out how to add autoattacks to the spell matches
+	core.MakeProcTriggerAura(&rogue.Unit, core.ProcTrigger{
+		Name:           label,
+		ClassSpellMask: ClassSpellMask_RogueDeadlyPoisonTick | ClassSpellMask_RogueOccultPoisonTick | ClassSpellMask_RogueInstantPoison,
+		Callback:       core.CallbackOnPeriodicDamageDealt | core.CallbackOnSpellHitDealt,
+		Outcome:        core.OutcomeLanded,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if spell.Matches(ClassSpellMask_RogueDeadlyPoisonTick|ClassSpellMask_RogueOccultPoisonTick|ClassSpellMask_RogueInstantPoison) && result.DidCrit() && sim.Proc(0.10, "Combo!") {
```

Comment:
And then for the proc chance, you can actually pass another option to the Proc Trigger config:
```
ProcChance: 0.10
```
and then remove this here.

Then you handler just becomes the line adding combo points